internal/common/common.go

```diff
@@ -178,6 +186,7 @@ func FromProvenance(prov *types.ValidatedProvenance) (*ProvenanceIR, error) {
 }
 
 // fromAmber maps data from a validated Amber provenance to ProvenanceIR.
+// invariant: for every data X in a validated Amber provenance that can be mapped to a field in `ProvenanceIR`, `fromAmber` sets a value v for X by using `WithX(v)`.
```

Comment:
Nice documentation :)

```suggestion
// invariant: for every data X in a validated Amber provenance that can be mapped to a field in `ProvenanceIR`, `fromAmber` sets a non-nil value v for X by using `WithX(v)`.
```

(or does "value" in Golang terminology imply non-nil?)

Reply:
Good point, added!